miden/tests/integration/main.rs

```diff
@@ -17,63 +17,95 @@ fn simple_program() {
+        $crate::build_test!(&source, $($tail)+)
     }};
-    ($source:expr, $stack_inputs:expr, $advice_tape:expr, $advice_sets:expr) => {{
-        let inputs =
-            $crate::helpers::ProgramInputs::new($stack_inputs, $advice_tape, $advice_sets).unwrap();
+}
 
-        $crate::helpers::Test {
-            source: String::from($source),
-            inputs,
-        }
-    }};
+/// Returns a Test struct in non debug mode from the provided source string, and any specified
+/// stack and advice inputs.
+///
+/// Parameters are expected in the following order:
+/// `source`, `stack_inputs` (optional), `advice_tape` (optional), `advice_sets` (optional)
+///
+/// * `source`: a well-formed source string.
+/// * `stack_inputs` (optional): the initial inputs which must be at the top of the stack before
+/// executing the `source`. Stack inputs can be provided independently without any advice inputs.
+/// * `advice_tape` (optional): the initial advice tape values. When provided, `stack_inputs` and
+/// `advice_sets` are also expected.
+/// * `advice_sets` (optional): the initial advice set values. When provided, `stack_inputs` and
+/// `advice_tape` are also expected.
+#[macro_export]
+macro_rules! build_test {
+    ($($params:tt)+) => {{
+        $crate::build_test_by_mode!(false, $($params)+)
+    }}
 }
 
-/// Returns a Test struct from a string of one or more operations and any specified stack and advice
-/// inputs.
+/// Returns a Test struct in debug mode from the provided source string and any specified stack
+/// and advice inputs.
 ///
 /// Parameters are expected in the following order:
 /// `source`, `stack_inputs` (optional), `advice_tape` (optional), `advice_sets` (optional)
 ///
-/// * `source`: a string of one or more operations, e.g. "push.1 push.2".
+/// * `source`: a well-formed source string.
 /// * `stack_inputs` (optional): the initial inputs which must be at the top of the stack before
 /// executing the `source`. Stack inputs can be provided independently without any advice inputs.
 /// * `advice_tape` (optional): the initial advice tape values. When provided, `stack_inputs` and
 /// `advice_sets` are also expected.
 /// * `advice_sets` (optional): the initial advice set values. When provided, `stack_inputs` and
 /// `advice_tape` are also expected.
 #[macro_export]
-macro_rules! build_op_test {
-    ($op_str:expr) => {{
-        let source = format!("begin {} end", $op_str);
-        $crate::build_test!(&source)
+macro_rules! build_debug_test {
+    ($($params:tt)+) => {{
+        $crate::build_test_by_mode!(true, $($params)+)
+    }}
+}
+
+/// This macro is an internal test builder, and is not intended to be called directly from tests.
+/// Instead, the `build_test` and `build_debug_test` wrappers should be used.
+#[macro_export]
+macro_rules! build_test_by_mode {
+    ($in_debug_mode:expr, $source:expr) => {{
+        $crate::helpers::Test::new($source, $in_debug_mode)
```

Comment:
nit: I would probably add a brief descriptor of this macro as well, even though I didn't in the code sample :) Maybe something like:

"Returns a Test struct in the specified debug or non-debug mode using the provided source string and any specified stack and advice inputs. Parameters start with a required boolean specifier for `in_debug_mode`, after which they match those of the `build_test` and `build_debug_test` macros. This macro is an internal test builder, and is not intended to be called directly from tests. Instead, the `build_test` and `build_debug_test` wrappers should be used."